### ISSUE TITLE
setLiveWallpaper doesn't work on xiaomi 10 miui 14

### ISSUE THREAD

**Issue by TDSSSzero** [codenameakshay/async_wallpaper#25]
**Describe the bug**
Everything works fine when I set up static images(jpg/png) on MIUI14，but when i use **liveWallpaper** ,both the lock screen and home screen changed back to the default wallpaper, and show toast successfully.
This may be a problem with MIUI itself, as I am still testing on Samsung and Huawei phones, and they are normal.
I want to know if there is a way to catch this exception and handle it.
sorry,my english not good,thank you watch this.

**code**
![image](https://github.com/codenameakshay/async_wallpaper/assets/75973660/a650f2e0-20a2-46b1-9430-f196393c118e)

**Screenshots**
![Screenshot_2023-07-10-11-13-01-545_tdsss save sav](https://github.com/codenameakshay/async_wallpaper/assets/75973660/43d87bf1-e2aa-42c7-a3f7-4198c492b88e)
![Screenshot_2023-07-10-11-16-16-841_com miui home](https://github.com/codenameakshay/async_wallpaper/assets/75973660/3ef65b1d-22e0-42c9-b346-2528a799be41)

**Flutter:**
 - version 3.7.11

**Dart:**
 - version 2.19.6

**Additional context**
phone: xiaomi 10 
android version Android 13 
MIUI 14.

**Comment by naeem-shah** [comment]
I have fix this issue and created a pull request modify your `pubspec.yaml`

  ```
async_wallpaper:
    git:
      url: https://github.com/codenameakshay/async_wallpaper.git
      ref: 10ae084c26d644a8c653aa0e63265d41a5603c8c
```

**Comment by codenameakshay** [comment]
The final target for a live wallpaper is controlled by Xiaomi/MIUI's system preview, so the plugin cannot portably force MIUI to preserve or replace only a specific screen, nor can it observe the user's final system choice as a normal Android API result.

PR #62 improves the part the plugin can control: video input is prepared atomically and validated before preview, plugin-owned success toasts are removed, and the result reports `previewOpened` / `awaitingUserConfirmation` instead of falsely claiming the wallpaper was applied. The older commit reference in this thread is superseded by PR #62.

Please test the 3.2.0 branch on the same Xiaomi 10 / MIUI 14 device:

```yaml
dependencies:
  async_wallpaper:
    git:
      url: https://github.com/codenameakshay/async_wallpaper.git
      ref: codex/3.2.0-wallpaper-engine
```

Then run:

```bash
flutter pub get
```

Please share the Android/MIUI build, options shown by the system preview, returned status, and whether the selected wallpaper persists after confirmation.

**Comment by codenameakshay** [comment]
Status after #66 (merged): this is a system/OEM behaviour, not something the plugin can fix in code. Android's live-wallpaper preview owns target selection and confirmation, and there is no callback telling the plugin whether the user accepted or cancelled, so "shows success then both screens revert" cannot be detected or rolled back by the plugin.

What did change: the plugin no longer claims success for these flows. `setVideoWallpaper` returns `awaitingUserConfirmation`, `openLiveWallpaperPreview` returns `previewOpened`, and `applied` is only reported for static wallpapers that Android actually confirmed per target. That distinction is documented in `doc/android-compatibility.md`, so an app can tell the user what the OS preview did rather than showing a success toast. Leaving this open as a tracked OEM limitation.
